ckb-importer/src/main/java/com/hartwig/serve/ckb/dao/TreatmentApproachDAO.java

```diff
@@ -42,20 +44,32 @@ public int write(@NotNull RelevantTreatmentApproaches treatmentApproaches) {
             writeTreatmentDrugClass(treatmentApproaches.drugClass(), id);
         }
 
+        if (treatmentApproaches.therapy() != null) {
+            writeTreatmentTherapy(treatmentApproaches.therapy(), id);
+        }
+
         for (Reference reference : treatmentApproaches.references()) {
             writeTreatmentReference(reference, id);
         }
 
         return id;
     }
 
-    private void writeTreatmentDrugClass(@Nullable DrugClass drugClass, int treatmentApproachDrugClassId) {
+    private void writeTreatmentDrugClass(@Nullable DrugClassInfo drugClassInfo, int treatmentApproachDrugClassId) {
         context.insertInto(Tables.TREATMENTAPPROACHDRUGCLASS,
                         Tables.TREATMENTAPPROACHDRUGCLASS.TREATMENTAPPROACHID,
                         Tables.TREATMENTAPPROACHDRUGCLASS.DRUGCLASSID,
-                        Tables.TREATMENTAPPROACHDRUGCLASS.CREATEDATE,
                         Tables.TREATMENTAPPROACHDRUGCLASS.DRUGCLASS)
-                .values(treatmentApproachDrugClassId, drugClass.id(), drugClass.createDate(), drugClass.drugClass())
+                .values(treatmentApproachDrugClassId, drugClassInfo.id(), drugClassInfo.drugClass())
+                .execute();
+    }
+
+    private void writeTreatmentTherapy(@Nullable TherapyInfo therapyInfo, int treatmentApproachDrugClassId) {
```

Comment:
Can this be @NotNull? Otherwise technically you need to check before extracting .id() etc. I think the only call site does a null check.